turbinia/lib/google_cloud.py

```diff
@@ -93,3 +98,44 @@ def setup_stackdriver_traceback(project_id):
     msg = 'Error enabling GCP Error Reporting: {0:s}'.format(str(exception))
     raise TurbiniaException(msg)
   return client
+
+
+def get_logs(output_dir, project_id, days, query=None):
+  """Copies stackdriver logs to a local directory.
+
+  Attributes:
+    project_id: The name of the Google Cloud project.
+    output_dir: The directory where logs are stored.
+    query: Query to use to pull stackdriver logs. 
+    days: number of days we want history for.
+  Raises:
+    TurbiniaException: When an error happens pulling the logs.
+  """
+  if not query:
+    query = 'jsonPayload.python_logger="turbinia"'
+  start_time = datetime.datetime.now() - timedelta(days=days)
+  start_string = start_time.strftime(DATETIME_FORMAT)
+  complete_query = '{0!s} timestamp>="{1!s}"'.format(query, start_string)
+  file_path = os.path.join(output_dir, "turbinia_stackdriver_logs.jsonl")
+  output_file = open(file_path, "w")
+  try:
+    client = cloud_logging.Client(project=project_id)
+    logger.info(
+        'Collecting the stackdriver logs with the following query: {}'.format(
+            complete_query))
+    logger.info('Writing the logs to {}'.format(file_path))
+    for entry in client.list_entries(order_by=cloud_logging.DESCENDING,
+                                     filter_=complete_query):
```

Comment:
Why do we need to set the filter here if that was part of the original query?

Reply:
Removing the filter will return all the stackdriver logs.

Reply:
Ah, of course.  For some reason I thought this had also been supplied earlier in another call.